File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.ts
function kidsWithCandies(candies: number[], extraCandies: number): boolean[] {
    const greatest: number = Math.max(...candies);
    let result: boolean[] = new Array(candies.length);

    for (let i: number = 0; i < candies.length; ++i) {
        if (candies[i] + extraCandies >= greatest) {
            result[i] = true;
        } else {
            result[i] = false;
        }
    }
    return result;
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int greatest = *max_element(candies.begin(), candies.end());
        vector<bool> result(candies.size());

        for (size_t i = 0; i < candies.size(); ++i) {
            if (candies[i] + extraCandies >= greatest) result[i] = true;
        }

        return result;
    }
};